input/azureeventhub/inputazureeventhub.go

```
@@ -0,0 +1,170 @@
+	if err != nil {
+		panic(err)
+	}
+
+	checkpointStore, err := checkpoints.NewBlobStore(containerClient, nil)
+	if err != nil {
+		panic(err)
+	}
+
+	wg := &sync.WaitGroup{}
+
+	go func() {
+		consumerClient, err := azeventhubs.NewConsumerClientFromConnectionString(t.EventHubNamespaceConnectionString, t.EventHub, t.ConsumerGroup, nil)
+		if err != nil {
+			panic(err)
+		}
+		defer consumerClient.Close(ctx)
+
+		var earliest bool = t.OffsetEarliest
+		var latest bool = !t.OffsetEarliest
+
+		processor, err := azeventhubs.NewProcessor(consumerClient, checkpointStore, &azeventhubs.ProcessorOptions{
+			UpdateInterval:        1 * time.Minute,
+			LoadBalancingStrategy: azeventhubs.ProcessorStrategyBalanced,
+			StartPositions: azeventhubs.StartPositions{
+				Default: azeventhubs.StartPosition{
+					Earliest: &earliest,
+					Latest:   &latest,
+				},
+			},
+		})
+		if err != nil {
+			panic(err)
+		}
+
+		go func() {
+			for {
+				partitionClient := processor.NextPartitionClient(ctx)
+				if partitionClient == nil {
+					break
+				}
+
+				go func() {
+					wg.Add(1)
+					defer partitionClient.Close(ctx)
+
+					for {
+						if ctx.Err() == context.Canceled {
+							break
+						}
+
+						receiveCtx, receiveCtxCancel := context.WithTimeout(ctx, time.Minute)
+						events, err := partitionClient.ReceiveEvents(receiveCtx, 100, nil)
+						receiveCtxCancel()
+
+						if err != nil && !errors.Is(err, context.DeadlineExceeded) {
+							if ctx.Err() != context.Canceled {
+								if eventHubError := (*azeventhubs.Error)(nil); errors.As(err, &eventHubError) && eventHubError.Code == azeventhubs.ErrorCodeOwnershipLost {
+									goglog.Logger.Debugln("partition rebalanced")
+								} else {
+									goglog.Logger.Errorf("error processing partition: %v", err)
+								}
+							}
+							break
+						}
+
+						for _, event := range events {
+							_, err = t.Codec.Decode(ctx, []byte(event.Body), t.Extras, []string{}, msgChan)
```

is it reasonable to ignore all errors?